### PR TITLE
Add horseshoe progress generation capabilities

### PR DESCRIPTION
**Pull request by mttr**
`horseshoe/?size=100&progress=75.png`
green

`horseshoe/?size=100&progress=27&color=red&.png`

red

`horseshoe/?size=282&progress=67&color=yellow&.png`

yellow

### CODE REVIEW

File: routes/horseshoe.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  ctx.strokeStyle = color;
+  ctx.stroke();
+}
+
+router.get('/', function(req, res) {
+  // default size to 200px and allow up to 2048px
+  var size = parseInt(req.query.size) || 200;
+  size = Math.max(Math.min(2048, +size), 1);
+
+  var progress = parseInt(req.query.progress);
```
Comment (diwu1989) [inline routes/horseshoe.js]: can you math min/max bound this progress guy between 0 and 100? for safety?

thanks

Comment (mttr) [inline routes/horseshoe.js]: take care of that in `renderProgress`

File: routes/horseshoe.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  var ctx = canvas.getContext('2d');
+
+  // fill the background
+  ctx.fillStyle = backgroundColor;
+  ctx.rect(0, 0, size, size);
+  ctx.fill();
+
+  var middle = Math.round(size * 0.5);
+
+  renderProgress(ctx, middle, baseColor, baseStrokeWidth, arcRadius, 100);
```
Comment (diwu1989) [inline routes/horseshoe.js]: base color is used only once, I think its okay to just inline it here

File: routes/horseshoe.js (hunk 1 above)
Comment (mttr) [inline routes/horseshoe.js]: oops. I do min it, but not max. hold on a minute.

File: routes/horseshoe.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  renderProgress(ctx, middle, progShadow, baseStrokeWidth, arcRadius, progress);
```
Comment (diwu1989) [inline routes/horseshoe.js]: the intermediate variables progShadow and progMainColor are only used once
if u used
`color.main` and `color.shadow` here, i think it'd be just as clear to the reader

File: app.js [hunk 4]
```diff
@@ -16,6 +18,7 @@ app.set('view engine', 'hbs');
 
 app.use(logger('dev'));
 app.use('/', routes);
+app.use('/horseshoe', require('./routes/horseshoe'));
```
Comment (diwu1989) [inline app.js]: for parity, i think u can make the other one
```js
app.use('/', require('./routes/index'));
app.use('/horseshoe', require('./routes/horseshoe'));

**Comment by diwu1989** [comment]
lgtm thanks!

**Pull request merged**